File: src/extension/scaffold/templates/universal-tool/handler.js
#!/usr/bin/env node
/**
 * Handler for {{name}} universal tool extension.
 *
 * Receives JSON on stdin, outputs JSON on stdout.
 * Expected input format:
 *     {"input": "..."}
 *
 * Output format:
 *     {"result": "...", "error": null}
 */

const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
  terminal: false,
});

rl.on('line', (line) => {
  try {
    const request = JSON.parse(line);
    const userInput = request.input || '';

    // TODO: Implement your tool logic here
    const result = `Processed: ${userInput}`;

    const response = { result, error: null };
    console.log(JSON.stringify(response));
  } catch (e) {
    const response = { result: null, error: e.message };
    console.log(JSON.stringify(response));
    process.exit(1);
  }
});

File: crates/extension-host/src/scaffold/templates/universal-tool/handler.py
#!/usr/bin/env python3
"""
Handler for {{name}} universal tool extension.

Receives JSON on stdin, outputs JSON on stdout.
Expected input format:
    {"input": "..."}

Output format:
    {"result": "...", "error": null}
"""

import json
import sys


def main():
    try:
        request = json.load(sys.stdin)
        user_input = request.get("input", "")

        # TODO: Implement your tool logic here
        result = f"Processed: {user_input}"

        response = {"result": result, "error": None}
        print(json.dumps(response))
    except Exception as e:
        response = {"result": None, "error": str(e)}
        print(json.dumps(response))
        sys.exit(1)


if __name__ == "__main__":
    main()
